forte/data/ontology/top.py

```diff
@@ -686,5 +688,169 @@ def get(
         yield from self.pack.get(entry_type, self, components, include_sub_type)
 
 
+class ImageAnnotation(Entry):
+    def __init__(self, pack: PackType, image_payload_idx: int):
+        """
+        ImageAnnotation type entries, such as "edge" and "bounding box".
+        Each ImageAnnotation has a ``image_payload_idx`` corresponding to its
+        image representation in the payload array.
+
+        Args:
+            pack: The container that this image annotation
+                will be added to.
+            image_payload_idx: A integer that represents the index of
+                the image in the payloads.
+        """
+        self._image_payload_idx = image_payload_idx
+        super().__init__(pack)
+
+    @property
+    def image_payload_idx(self) -> int:
+        return self._image_payload_idx
+
+    @property
+    def image(self):
+        if self.pack is None:
+            raise ValueError(
+                "Cannot get image because image annotation is not "
+                "attached to any data pack."
+            )
+        return self.pack.get_image_array(self._image_payload_idx)
+
+    def __eq__(self, other):
+        if other is None:
+            return False
+        return self.image_payload_idx == other.image_payload_idx
+
+    def __hash__(self) -> int:
+        r"""
+        The hash function for ``Sketch`` class with a numpy array as a class
+        attribute.
+        """
+        return hash((self._image_payload_idx, self._tid))
+
+    @property
+    def index_key(self) -> int:
+        return self.tid
+
+
+class Grids(Entry):
+    """
+    Regular grids with a grid configuration.
+
+    Args:
+        pack: The container that this grids
+            will be added to.
+        height_n_width: A tuple represents the number of grid cell per column and
+            the number of grid cell per row.
+    """
+
+    def __init__(
+        self,
+        pack: PackType,
+        height: int,
+        width: int,
+        image_payload_idx: Optional[int] = None,
+    ):
+
+        if height <= 0 or width <= 0:
+            raise ValueError(
+                f"height({height}) and "
+                f"width({width}) both must be larger than 0"
+            )
+        self.height = height
```

Comment:
Could be private, then provide a property to access it?
```suggestion
        self._height = height
```